cores/esp8266/core_esp8266_waveform.h

```diff
@@ -50,6 +50,8 @@ extern "C" {
 // If runtimeUS > 0 then automatically stop it after that many usecs.
 // Returns true or false on success or failure.
 int startWaveform(uint8_t pin, uint32_t timeHighUS, uint32_t timeLowUS, uint32_t runTimeUS);
+// Same as above, but pass in CPU clock cycles instead of microseconds
```

Comment:
I suggest adding a comment here that this form has a resolution problem above 977Hz: the higher the frequency the lower the step resolution, until at 40KHz there are only 23 (is that correct?) discrete values possible for the duty cycle.

Reply:
That's actually a PWM limitation, not the waveform one.  The analogWrite suffers from that, but if you're calling the bare API with a "us" unit, that's pretty obvious what the granularity is. :)

In analogWrite, we're already fixed it so the comment isn't valid anymore.